Update auth-sds user guide (#4901)
istio/istio.io: content/en/docs/tasks/security/auth-sds/index.md

Currently, the SDS identity provision flow has the following caveats:

> * The SDS feature is currently in Alpha phase.

Reviewer: Add link to the feature status page:
```suggestion
* SDS support is currently in [Alpha](/about/feature-stages/#security-and-policy-enforcement).
```

> * SDS for the control plane security is enabled by default, and it is alpha version.

Reviewer 2: ```suggestion
* The SDS feature is currently in Alpha phase.
```

Reviewer 2: I don't think we need to mention too much about the control plane security. Our users should naturally think the control plane security is using SDS.
